R/cv_varsel_formula.R

```diff
@@ -0,0 +1,370 @@
+	  as.search <- relax
+    submodels <- .get_submodels_poc(spath, c(0, seq_along(vind)), fam, p_pred,
+                                    refmodel, intercept, opt$regul, as.search=as.search)
+		summaries_sub <- .get_sub_summaries_poc(submodels, c(i), refmodel, fam)
+
+    for (k in seq_along(summaries_sub)) {
+      loo_sub[i,k] <- summaries_sub[[k]]$lppd
+      mu_sub[i,k] <- summaries_sub[[k]]$mu
+    }
+
+    ## we are always doing group selection
+    ## with `match` we get the indices of the variables as they enter the
+    ## solution path in vind
+    vind_mat[i,] <- match(vind, groups)
+
+    if (verbose) {
+		  utils::setTxtProgressBar(pb, run_index)
+		}
+	}
+
+	if (verbose)
+	  # close the progress bar object
+	  close(pb)
+
+	# put all the results together in the form required by cv_varsel
+	summ_sub <-	lapply(0:nv_max, function(k){
+	    list(lppd=loo_sub[,k+1], mu=mu_sub[,k+1], w=validset$w)
+	})
+	summ_ref <- list(lppd=loo_ref, mu=mu_ref)
+	summaries <- list(sub=summ_sub, ref=summ_ref)
+
+  vind_cv <- lapply(1:n, function(i){ vind_mat[i,] })
+
+  d_test <- list(y=refmodel$y, type='loo',
+                 test_points=seq_along(refmodel$y))
+
+	return(list(vind_cv=vind_cv, summaries=summaries, d_test=d_test))
+
+}
+
+kfold_varsel_poc <- function(refmodel, method, nv_max, ns, nc, nspred, ncpred, relax,
+                             intercept, penalty, verbose, opt, K, seed=NULL, groups=NULL) {
+
+  group_features <- !is.null(groups)
+	# fetch the k_fold list (or compute it now if not already computed)
+	k_fold <- .get_kfold_poc(refmodel, K, verbose, seed)
+
+	# check that k_fold has the correct form
+	# .validate_kfold(refmodel, k_fold, refmodel$nobs)
+
+	K <- length(k_fold$refmodel)
+	family_kl <- refmodel$fam
+
+  # extract variables from each fit-object (samples, x, y, etc.)
+  # to a list of size K
+	refmodels_cv <- k_fold$refmodel
+
+  # List of K elements, each containing d_train, p_pred, etc. corresponding
+  # to each fold.
+  msgs <- paste0(method, ' search for fold ', 1:K, '/', K, '.')
+  list_cv <- mapply(function(refmod, test_points, msg) {
+  	p_sel <- .get_refdist(refmod, ns, nc)
+  	p_pred <- .get_refdist(refmod, nspred, ncpred)
+    newdata <- refmod$fetch_data(data_points=test_points)
+  	mu_test <- refmod$predfun(refmod$fit, newdata=newdata)
+  	list(refmodel = refmod, p_sel = p_sel, p_pred = p_pred,
+         mu_test = mu_test, dis = refmod$dis, w_test = refmod$wsample,
+         msg = msg, y_test = refmod$y[test_points], test_points = test_points)
+  }, refmodels_cv, k_fold$test_points, msgs, SIMPLIFY = F)
+
+  ## Perform the selection for each of the K folds
+  if (verbose) {
+    print('Performing selection for each fold..')
+    pb <- utils::txtProgressBar(min = 0, max = K, style = 3, initial=0)
+  }
+  spath_cv <- lapply(seq_along(list_cv), function(fold_index) {
+    fold <- list_cv[[fold_index]]
+    family_kl <- fold$refmodel$family
```

Comment:
It was defined as `fam` in the current version, I used `family` but didn't change it in all places. Now I've renamed it everywhere to `family`. I think this renaming should be applied as well to `family_kl`, because the `kl` part is quite specific and it's possible that in future versions we play with other divergences or distances?